var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -34,65 +40,106 @@ class Mesa(AutotoolsPackage):
     list_depth = 2
 
+    version('17.1.5', '6cf936fbcaadd98924298a7009e8265d')
+    version('17.1.4', 'be2ef7c9edec23b07f74f6512a6a6fa5')
     version('17.1.3', '1946a93d543bc219427e2bebe2ac4752')
+    version('17.1.1', 'a4844bc6052578574f9629458bcbb749')
     version('13.0.6', '1e5a769bc6cfd839cf3febcb179c27cc')
     version('12.0.6', '1a3d4fea0656c208db59289e4ed33b3f')
     version('12.0.3', '1113699c714042d8c4df4766be8c57d8')
 
-    variant('swrender', default=False,
-        description="Build with (gallium) software rendering.")
-
+    variant('swrender', default=True,
+            description="Build with (gallium) software rendering.")
+    variant('hwrender', default=False,
+            description="Build with (DRI) hardware rendering.")
     variant('llvm', default=False,
-        description="Use llvm for rendering pipes.")
+            description="Use llvm for rendering pipes.")
 
     # General dependencies
-    depends_on('python@2.6.4:')
-    depends_on('py-mako@0.3.4:', type=('build', 'run'))
+    depends_on('pkg-config@0.9.0:', type='build')
     depends_on('flex@2.5.35:', type='build')
     depends_on('bison@2.4.1:', type='build')
-    depends_on('pkg-config@0.9.0:', type='build')
-
-    # Off-screen with llvmpipe
-    # Note: there must be a better way of selecting the preferred llvm
-    depends_on('llvm+link_dylib', when='+llvm')
-
-    # For DRI and hardware acceleration
+    depends_on('binutils', type='build')
+    depends_on('python@2.6.4:', type='build')
+    depends_on('py-mako@0.3.4:', type='build')
+    depends_on('gettext')
+    depends_on('icu4c')
+    depends_on('expat')
     depends_on('libpthread-stubs')
-    depends_on('libdrm')
     depends_on('openssl')
+    depends_on('xproto')
+    depends_on('glproto@1.4.14:')
+    depends_on('presentproto@1.0:')
     depends_on('libxcb@1.9.3:')
-    depends_on('libxshmfence@1.1:', when='~swrender')
-    depends_on('libx11', when='~swrender')
-    depends_on('libxext', when='~swrender')
-    depends_on('libxdamage', when='~swrender')
+    depends_on('libx11')
+    depends_on('libxext')
+    depends_on('libxshmfence@1.1:')
+    depends_on('libxdamage')
     depends_on('libxfixes')
-    # depends_on('expat', when='~swrender')
-    depends_on('libelf', when='+llvm~swrender')
+    depends_on('libxv')
+    depends_on('libxvmc')
 
-    depends_on('glproto@1.4.14:', type='build', when='~swrender')
-    depends_on('dri2proto@2.6:', type='build', when='~swrender')
-    depends_on('dri3proto@1.0:', type='build', when='~swrender')
-    depends_on('presentproto@1.0:', type='build', when='~swrender')
+    # For DRI and hardware acceleration
+    depends_on('dri2proto@2.6:', type='build')
+    depends_on('dri3proto@1.0:', type='build')
+    depends_on('libdrm')
 
-    # TODO: Add package for systemd, provides libudev
-    # Using the system package manager to install systemd didn't work for me
+    # Note: there must be a better way of selecting the preferred llvm
```

Comment:
Nope, this is the correct way. You should also add:
```python
depends_on('llvm+link_dylib', when='+llvm')
```
Otherwise `spack install mesa ^llvm@1.2.3` won't work because "mesa does not depend on llvm".